### PR TITLE
Temporal: Move all tests out of total.js under staging into separate files under test/built-ins

### PR DESCRIPTION
**Pull request by catamorphism**
Break up `Temporal/Duration/old/total.js` into separate tests and add those files under `test/built-ins`.

### CODE REVIEW

File: test/built-ins/Temporal/Duration/prototype/total/relativeto-not-required-to-round-fixed-length-units-in-durations-without-variable-units.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+// relativeTo not required to round fixed-length units in durations without variable units (negative)
+const negativeD2 = d2.negated();
+assert(Math.abs(negativeD2.total({ unit: "days" }) - -totalD2.days) < Number.EPSILON);
+assert(Math.abs(negativeD2.total({ unit: "hours" }) - -totalD2.hours) < Number.EPSILON);
+assert(Math.abs(negativeD2.total({ unit: "minutes" }) - -totalD2.minutes) < Number.EPSILON);
+assert(Math.abs(negativeD2.total({ unit: "seconds" }) - -totalD2.seconds) < Number.EPSILON);
+assert(Math.abs(negativeD2.total({ unit: "milliseconds" }) - -totalD2.milliseconds) < Number.EPSILON);
+assert(Math.abs(negativeD2.total({ unit: "microseconds" }) - -totalD2.microseconds) < Number.EPSILON);
+assert.sameValue(negativeD2.total({ unit: "nanoseconds" }), -totalD2.nanoseconds);
```
Comment (ptomato) [inline test/built-ins/Temporal/Duration/prototype/total/relativeto-not-required-to-round-fixed-length-units-in-durations-without-variable-units.js:39]: Same here as in accepts-datetime-strings-or-fields-for-relativeto.js — we probably want to determine the exact mathematical value, then make sure we have the correct f64 representation of it.

(Likewise, OK to consider this out of scope and stick this test back in staging for the time being.)

File: test/built-ins/Temporal/Duration/prototype/total/balances-differently-depending-on-relativeto.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+description: Balances differently depending on relativeTo.
+features: [Temporal]
+---*/
+
+const fortyDays = new Temporal.Duration(0, 0, 0, 40, 0, 0, 0, 0, 0, 0);
+const negativeFortyDays = new Temporal.Duration(0, 0, 0, -40, 0, 0, 0, 0, 0, 0);
+
+assert.sameValue(fortyDays.total({
+  unit: "months",
+  relativeTo: "2020-02-01"
```
Comment (ptomato) [inline test/built-ins/Temporal/Duration/prototype/total/balances-differently-depending-on-relativeto.js:15]:
```suggestion
  relativeTo: new Temporal.PlainDate(2020, 2, 1)
```
(and elsewhere in this file)

File: test/built-ins/Temporal/Duration/prototype/total/balances-differently-depending-on-relativeto.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+}).toPrecision(16), (1 + 11 / 31).toPrecision(16));
```
Comment (ptomato) [inline test/built-ins/Temporal/Duration/prototype/total/balances-differently-depending-on-relativeto.js:16]: Also, same comment about exact mathematical values. Likewise, fine to stick this one back in staging for the time being.

File: test/built-ins/Temporal/Duration/prototype/total/balances-up-to-next-unit-after-rounding.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+---*/
+
+const almostWeek = new Temporal.Duration(0, 0, 0, 6, 20, 0, 0, 0, 0, 0);
+const almostWeekNegative = new Temporal.Duration(0, 0, 0, -6, -20, 0, 0, 0, 0, 0);
+
+const totalWeeks = almostWeek.total({
+  unit: "weeks",
+  relativeTo: "2020-01-01"
+});
+assert(Math.abs(totalWeeks - (6 + 20 / 24) / 7) < Number.EPSILON);
```
Comment (ptomato) [inline test/built-ins/Temporal/Duration/prototype/total/balances-up-to-next-unit-after-rounding.js:17]: Same two comments as previous file.

File: test/built-ins/Temporal/Duration/prototype/total/balances-days-up-to-both-years-and-months.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+/*---
+esid: sec-temporal.duration.prototype.total
+description: Balances days up to both years and months.
+features: [Temporal]
+---*/
+
+const twoYears = new Temporal.Duration(0, 11, 0, 396, 0, 0, 0, 0, 0, 0);
+assert.sameValue(twoYears.total({
+  unit: "years",
+  relativeTo: "2017-01-01"
```
Comment (ptomato) [inline test/built-ins/Temporal/Duration/prototype/total/balances-days-up-to-both-years-and-months.js:13]:
```suggestion
  relativeTo: new Temporal.PlainDate(2017, 1, 1)
```
(same below)

**Review by ptomato — APPROVED** [review]
Looks good. Unfortunately at a long-ago stage in the proposal's evolution, I wrote some of these less rigorously regarding mathematical values and floating-point values than they should be for test262. Fixing that seems like a bigger chunk of work than is warranted for this PR, so it'd be fine with me to leave those particular tests in staging for the time being, and one of us can do them in a follow up.

File: test/built-ins/Temporal/Duration/prototype/total/relativeto-not-required-to-round-fixed-length-units-in-durations-without-variable-units.js (hunk 1 above)
Comment (catamorphism) [reply test/built-ins/Temporal/Duration/prototype/total/relativeto-not-required-to-round-fixed-length-units-in-durations-without-variable-units.js]: Moved back into staging

File: test/built-ins/Temporal/Duration/prototype/total/balances-differently-depending-on-relativeto.js (hunk 2 above)
Comment (catamorphism) [reply test/built-ins/Temporal/Duration/prototype/total/balances-differently-depending-on-relativeto.js]: Done

File: test/built-ins/Temporal/Duration/prototype/total/balances-up-to-next-unit-after-rounding.js (hunk 4 above)
Comment (catamorphism) [reply test/built-ins/Temporal/Duration/prototype/total/balances-up-to-next-unit-after-rounding.js]: Fixed relativeTo and moved back into staging

File: test/built-ins/Temporal/Duration/prototype/total/balances-days-up-to-both-years-and-months.js (hunk 5 above)
Comment (catamorphism) [reply test/built-ins/Temporal/Duration/prototype/total/balances-days-up-to-both-years-and-months.js]: Done

**Pull request merged**